misc/include/naive_rmm_tree.h

```diff
@@ -0,0 +1,274 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <cstdint>
+#include <limits>
+#include <algorithm>
+
+struct NaiveRmM
+{
+    using u64 = std::uint64_t;
+    static constexpr std::size_t npos = std::numeric_limits<std::size_t>::max();
```

Comment:
Yet again, the name is non-informative for the reader. At least something like `maximum_position`

Reply:
In this case `npos` is ok as it is described right now. Small note on the usage of `std`: namespace `std` is by far the most used namespace and a naming guide for other so name collisions with the std is of big concern. For this reason it is a bad practice to do `using namespace std` or even `using std::<whatever>`. Doing something like `using my_aweome_type = std::<whatever>` is fine.

`std::optional` is good but I'm not sure if it is free in terms of overhead, usage on low-level primitives i questionable. A good thing would be to just test two implementations with and without `std::optional` to check runtime difference.